optuna/trial/_trial.py

```diff
@@ -557,7 +557,7 @@ def objective(trial):
 
         self.storage.set_trial_user_attr(self._trial_id, key, value)
 
-    def set_system_attr(self, key: str, value: Any) -> None:
+    def _set_system_attr(self, key: str, value: Any) -> None:
```

Comment:
Could you avoid renaming `set_system_attr` to `_set_system_attr` since it's a breaking change. We should just deprecate it (by wrapping with a `@deprecated` decorator) and then update the docstring.

Reply:
I added @deprecated_func("3.1.0", "6.0.0") to `set_system_attr` method.

Reply:
For now, I set the timing of deprecation `"6.0.0"`. Is it reasonable choice? Or too far?

Reply:
Sounds reasonable 👍 It follows our deprecation policy.
https://github.com/optuna/optuna/wiki/Deprecation-policy

Reply:
By default, we will remove it in v5.0.0 (two major versions after v3.1.0). But I agree to defer the version because users seem to use it, and the maintenance cost is low.